### DISCUSSION TITLE
Errors after upgrade to Nuxt 3.9.2 "Importing from server is not allowed in the Vue part of your app."

### DISCUSSION THREAD

**Discussion by ivan-mihalic** [nuxt/nuxt#25534] [Questions] [answered]
Hi guys,
I have just upgraded to nuxt v 3.9.2 (thought 3.10.0) and these errors started display in terminal.
App seems to run ok.
Does anyone facing this same issue?

for example in `server/api/health.get.ts` I have only `defineEventHandler` that returns 'OK'

```
export default defineEventHandler(async () => {
  return 'OK';
});
```

I have error for all `.ts` files in `server/api`, `server/middleware` and `server/plugins` folders. Here is example of few errors.

```
[3:33:12 PM]  ERROR  Importing from server is not allowed in the Vue part of your app. [importing ../../../../../server/api/health.get.ts from src/index.html]


[3:33:12 PM]  ERROR  Importing from server is not allowed in the Vue part of your app. [importing ../../../../../server/middleware/morgan.ts from src/index.html]


[3:33:12 PM]  ERROR  Importing from server is not allowed in the Vue part of your app. [importing ../../../../../server/plugins/storage.ts from src/index.html]


[3:33:12 PM]  ERROR  Importing from server is not allowed in the Vue part of your app. [importing ../../../../../server/api/ares/index.post.ts from src/index.html]
```

**Comment by tclxshuai-lv** [comment]
same error message and the application worked fine, but I still downgraded the nuxt to version 3.9.1

**Reply by TheAlexLichter** [reply]
This is no Nuxt issue but an issue with your code or a module that you are using :)

**Comment by noriyuki-shimizu** [comment]
@ivan-mihalic @tclxshuai-lv
I have an error with the following directory structure.
````
root
├──...
├──store
│   └── page
│       └── articles
│           ├── qiita.ts
│           └── zenn.ts (<- import "~/infrastructures/rest/nuxt-server/api/zenn/articles")
└── infrastructures
    └── rest
        ├── nuxt-server
        │   └── api
        │       └── zenn
        │           └── articles.ts
        └── qiita.com
            └── api
                └── v2
                    └── items.ts
````

The error message is as follows.
```
Error:  Importing from server is not allowed in the Vue part of your app. [importing `infrastructures/rest/nuxt-server/api/zenn/articles` from `store/page/articles/zenn.ts`]
```

As a fix, I changed the directory name "nuxt-server" to "nuxt" and the above error went away.
I am assuming that importing a file that contains the directory name "server" into a file used by a Vue application would cause the error.

If this is a completely different case, this answer may not be helpful, but I hope it can be of some help.

Target repository: https://github.com/noriyuki-shimizu/article-nuxt3

**Reply by TheAlexLichter** [reply]
Yes, that might be indeed a naming-thing!

**Comment by sifferhans** [comment]
I face the same issue! This should probably be an issue, and not a discussion, though.

**Reply by TheAlexLichter** [reply]
This is no Nuxt issue but an issue with your code or a module that you are using :)

**Comment by TheAlexLichter** [comment]
@ivan-mihalic You are using code during runtime which is not meant to be used during runtime. TL;DR - Nuxt is warning you that you are misusing something.

Often it is only the logger imported from @nuxt/kit. As replacement, just use consola.

If it is something else, share the code 👍 

---

In your case, what is `src/index.html`?

**Reply by ivan-mihalic** [reply]
I have no src/index.html file

**Comment by sifferhans** [comment]
@manniL I have the same problem, and I am not importing anything from the server in any Vue files. Also, there is not `src/index.html` in my codebase at all 🤔

**Reply by TheAlexLichter** [reply]
Can you share the whole error message?

**Reply by tclxshuai-lv** [reply]
@manniL  My project doesn't have an index.html  ![image](https://github.com/nuxt/nuxt/assets/150904574/11f46792-d77c-405c-bd58-ee1c87621bb2)

**Comment by ivan-mihalic** [comment]
I am digging. but I have no solution yet.
I found out that it started from 3.9.2. It looks like `import-protection` causes it.

[https://github.com/nuxt/nuxt/pull/25162/files/b4efe1cceca2cad78610f74ceda006f38df2240e#diff-72254fc84e6c48d91f79e2bc60dbf0fbcbb80d4fabd9d1b05598acf4cf10602d](https://github.com/nuxt/nuxt/pull/25162/files/b4efe1cceca2cad78610f74ceda006f38df2240e#diff-72254fc84e6c48d91f79e2bc60dbf0fbcbb80d4fabd9d1b05598acf4cf10602d)

**Reply by ivan-mihalic** [reply, marked as answer]
I think I found it. 
I have srcDir = ./src in nuxt.config. 

I have server folder within src folder (./src/server). 

when I put server folder on src level and add serverDir = ./server

errors are gone

**Comment by angelhdzmultimedia** [comment]
`Importing from "X" is not allowed in the Vue part of your app.`
I got this error for first time when I installed a Vue Plugin `~/modules/api/runtime/api.plugin.ts`, that used an object (authApi) exported from the index.ts of a module (~/modules/auth), which imports members of the 'nuxt/kit' package.

Apparently `nuxt/kit` or `@nuxt/kit` package is not available in the Vue part of the app, and it makes sense, but until I understood what was happening I was confused. So in my case, I moved `authApi` object (which contains an instance of AuthController from `~/modules/auth/auth.controller.ts`) to `~/modules/auth/auth.config.ts`, and because there I'm not using anything from 'nuxt/kit' the error went away.

**Comment by tclxshuai-lv** [comment]
@ivan-mihalic Did you install @nuxtjs/tailwindcss？ If i delete the @nuxtjs/tailwindcss package, the error message disappears

**Reply by ivan-mihalic** [reply]
I solved issue with moving server dir outside the src folder

**Comment by tclxshuai-lv** [comment]
@ivan-mihalic @manniL I create a minimal reproduction：https://stackblitz.com/edit/github-hoyhk3?file=package.json

**Reply by TheAlexLichter** [reply]
~This is indeed a bug when using tailwindcss + srcDir it seems 👍~

When adding `./src/**/*.XYZ` to `content`, the tailwind plugin will import and analyze also the `server/` files, triggering the protection

**Reply by TheAlexLichter** [reply]
[Reported here](https://github.com/nuxt-modules/tailwindcss/issues/811)

**Reply by ineshbose** [reply]
[Answered](https://github.com/nuxt-modules/tailwindcss/issues/811#issuecomment-1968424509) (feel free to continue discussion there)
